Sources/SPTDataLoader.m

```diff
@@ -240,9 +294,17 @@ - (void)receivedInitialResponse:(SPTDataLoaderResponse *)response
     }
     
     if ([self.delegate respondsToSelector:@selector(dataLoader:didReceiveInitialResponse:)]) {
-        [self executeDelegateBlock: ^{
+        [self executeDelegateBlock:^{
             [self.delegate dataLoader:self didReceiveInitialResponse:response];
         }];
+    } else {
+        [self executeDelegateBlock:^{
+            void (^handler)(SPTDataLoaderResult *) = [self.requestHandlers objectForKey:response.request];
+            handler([SPTDataLoaderResult fromResponse:response kind:SPTDataLoaderResultReceivedInitialResponse]);
```

Comment:
Always check for `nil` before invoking a block. If `handler == nil`, this will crash.